### DISCUSSION TITLE
iOS: swipe back gesture

### DISCUSSION THREAD

**Discussion by auseika** [ionic-team/capacitor#3137] [Q&A] [answered]
### Capacitor Version
  @capacitor/cli 2.2.0

  @capacitor/android 2.2.0

  @capacitor/ios 2.2.0

  @capacitor/core 2.2.0

### Platform(s)
iOS

### Current Behavior
Basically, I'm unable to use swipe back gesture to navigate, which is an essential thing for ios.

### Other Technical Details
We have a nextjs app exported to statics, which runs well after wrapping it inside Capacitor. However, I can't use back swipe plugin, which crashes Android and also feels not responsive enough on iOS.

What should I do?

**Comment by imhoffd** [comment, marked as answer]
If I'm understanding correctly, your app is compiled into many HTML files and you're trying to use native web view back swipe to navigate back? If that's correct, I would [create a proposal for a community plugin](https://github.com/capacitor-community/proposals/) (it's okay if it exists as third-party already, just link to it).

**Reply by auseika** [reply]
Thanks.
So, it shouldn't work out of the box, thats a relief.

Current backswipe plugin is not maintained for almost a year, crashes Android build without modification, and also not relable at all on ios. Will add a proposal.

**Reply by imhoffd** [reply]
Awesome! Yeah, Capacitor mainly assumes you're using a SPA. Additionally, mobile UI frameworks provide a web implementation of back swipe. I think we want to keep it as is for now. A back swipe plugin would be a great addition as a community plugin.

Thanks!

**Reply by rashad** [reply]
Has there been any updates on this subject in the last year by any chance?

**Reply by ollyde** [reply]
Has anyone implemented this? I can only find "capacitor-swipe-back" and it's broken..

**Reply by pferreirafabricio** [reply]
> Has there been any updates on this subject in the last year by any chance?

I was able to resolve this issue by following these steps:

1. Create a [custom view controller](https://capacitorjs.com/docs/ios/viewcontroller)
2. Use [this code](https://github.com/ionic-team/capacitor/issues/1097#issuecomment-1077649377)

**OBS:** When testing "_your mouse cursor needs to be exactly on the edge that you're swiping from or just beyond_", more on: https://github.com/ionic-team/ionic-framework/issues/25781#issuecomment-1220858882

**Reply by ollyde** [reply]
@pferreirafabricio it doesn't work when going forwards?

**Reply by pferreirafabricio** [reply]
> @pferreirafabricio it doesn't work when going forwards?

@ollyde I have tested going forward and seems to be okay. Are you having any trouble?

**Reply by ollyde** [reply]
@pferreirafabricio sorry for the late reply.

I attached a video, you can see two bugs here.

1. Going back, IE popping the stack does not result in a view animation.
2. Going forwards does not result in a view animation.

The only thing that works is swiping back; this is a poor result for iOS and Android.

https://user-images.githubusercontent.com/6940726/232299029-2fd9b196-cd4e-4c3c-b74e-5d7db7b323b2.mp4

**Comment by lazysergey** [comment]
this solved `iOS` swiping for me
https://georghoeller.dev/capacitor-ng-back-swipe/amp/

**Reply by ThatOneCalculator** [reply]
Came looking for this and their site is down, thankfully an archive is here: https://web.archive.org/web/20240722195047/https://georghoeller.dev/capacitor-ng-back-swipe/
